src/test/java/teammates/test/cases/ui/browsertests/AdminSessionsPageUiTest.java

```diff
@@ -63,13 +63,21 @@ private void testContent() {
         ______TS("content: hide filter");
         
         sessionsPage.clickDetailButton();
-        assertFalse(isTimeFramePanelVisible());
+        assertTrue(isTimeFramePanelInvisible());
         assertTrue(isSessionDataDisplayCorrect());
         
     }
     
     private boolean isTimeFramePanelVisible() {
-        return sessionsPage.isElementVisible(By.id("timeFramePanel"));
+        By timeFramePanel = By.id("timeFramePanel");
+        sessionsPage.waitForElementPresence(timeFramePanel);
+        return sessionsPage.isElementVisible(timeFramePanel) == true;
+    }
+
+    private boolean isTimeFramePanelInvisible() {
+        By timeFramePanel = By.id("timeFramePanel");
+        sessionsPage.waitForElementToDisappear(timeFramePanel);
+        return sessionsPage.isElementVisible(timeFramePanel) == false;
```

Comment:
You don't actually have to do the `isElementVisible` comparison anymore. If the element doesn't disappear in time, it will throw a `TimeoutException`.